File: ios/Calendar.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTLog.h>
#import "Calendar.h"

#import <React/RCTEventEmitter.h>


@interface RCT_EXTERN_MODULE(Calendar, NSObject)
RCT_EXTERN_METHOD(addEventToCalendar: (NSString *)titleEvent
                  startDate: (nonnull NSNumber)startDate
                  endDate: (nonnull NSNumber)endDate
                  resolve:(RCTPromiseResolveBlock) resolve
                  reject: (RCTPromiseRejectBlock)reject)
@end

File: ios/Calendar.swift
import Foundation
import EventKit
import EventKitUI
import React

@objc(Calendar)
class Calendar: NSObject
{
  @objc(addEventToCalendar:startDate:endDate:resolve:reject:)
  func addEventToCalendar(_ titleEvent: String, startDate: NSNumber, endDate: NSNumber, resolve:  @escaping RCTPromiseResolveBlock, reject:  @escaping RCTPromiseRejectBlock) {
    let eventStore = EKEventStore();
    
    eventStore.requestAccess(to: .event, completion: {(granted, error) in
      if (granted) && (error == nil) {
        let event: EKEvent = EKEvent(eventStore: eventStore)
        event.title = titleEvent
        event.startDate = Date()
        event.endDate = Date()
        event.notes = "This is a note"
        event.calendar = eventStore.defaultCalendarForNewEvents
        
        do {
          try eventStore.save(event, span: .thisEvent);
          resolve("Событие успешно добавлено в календарь");
        } catch let err as NSError{
          reject("Error", "Не удалось добавить событие в календарь", err);
        }
      } else {
        reject("Permission_Error", "Не удалось получить доступ к календарю", error);
      }
    })
  }
}
